builder/remotecontext/git/gitutils.go

```diff
@@ -213,7 +213,7 @@ func (repo gitRepo) gitWithinDir(dir string, args ...string) ([]byte, error) {
 	cmd := exec.Command("git", args...)
 	cmd.Dir = dir
 	// Disable unsafe remote protocols.
-	cmd.Env = append(cmd.Env, "GIT_PROTOCOL_FROM_USER=0")
+	cmd.Env = append(cmd.Environ(), "GIT_PROTOCOL_FROM_USER=0")
```

Comment:
`(*exec.Cmd).Environ()` is new in Go 1.19. Hope that's okay, as `os.Environ()` is not an exact drop-in replacement on Windows: see https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/internal/syscall/execenv/execenv_windows.go

Reply:
Yes, we can. `os.Environ()` _is_ in fact an exact drop-in replacement in this case as the diverging Windows code path is only taken when `cmd.SysProcAttr` is set.